sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/RecordWriter.java

```diff
@@ -69,7 +69,8 @@ class RecordWriter {
     }
     OutputFile outputFile;
     try (FileIO io = table.io()) {
-      outputFile = io.newOutputFile(absoluteFilename);
+      OutputFile tmpFile = io.newOutputFile(absoluteFilename);
+      outputFile = (OutputFile) table.encryption().encrypt(tmpFile);
```

Comment:
Note that this line will actually encrypt the outputFile and wrap it in a [`EncryptedOutputFile`](https://github.com/apache/iceberg/blob/a50ec923f3d928f67e2a4a361c0d1162341aa084/api/src/main/java/org/apache/iceberg/encryption/EncryptedOutputFile.java#L30). This object holds two key pieces:
- `.encryptingOutputFile()`: the wrapped OutputFile that is primed for encryption 
- `.keyMetadata()`: the information needed to encrypt the file when writing

So we should actually retrieve the wrapped output file, and pass the keyMetadata piece to our writers below, e.g.

```java
  .withKeyMetadata(keyMetadata)
```